nft-client/src/main.rs

```diff
@@ -0,0 +1,635 @@
+//! This is an example of an off-chain application which integrates with a smart
+//! contract on chain. The application is a client for working with a specific
+//! implementation of a Non-fungible token(NFT) smart contract.
+//! The smart contract should follow the CTS1 specification, but some additional
+//! assumptions are made for how the contract serializes the state and the
+//! existence of a "mint" contract function.
+//!
+//! The smart contract is called "CTS1-NFT" and can be found [here](https://github.com/Concordium/concordium-rust-smart-contracts/tree/main/examples).
+//!
+//! Most of the functionality needs access to the GRPC API of a running
+//! concordium-node. Some functionality of the client additionally depends on
+//! the concordium-node to have the transaction logging enabled and access to
+//! the PostgreSQL database with transaction logs.
+
+use anyhow::*;
+use clap::AppSettings;
+use concordium_rust_sdk::{
+    common, constants,
+    endpoints::{Client, Endpoint},
+    id, postgres,
+    postgres::DatabaseSummaryEntry,
+    types,
+};
+use futures::{StreamExt, TryStreamExt};
+use serde::*;
+use smart_contracts::concordium_contracts_common;
+use std::{
+    collections::{BTreeMap as Map, BTreeSet as Set},
+    convert::TryFrom,
+    fmt::Display,
+    path::PathBuf,
+    str::FromStr,
+};
+use structopt::*;
+use thiserror::*;
+use types::{smart_contracts, transactions};
+mod cts1;
```

Comment:
Please don't mix `mod` and `use`. 

Also, please have `lib.rs` instead of declaring the module `cts1` in the `main` module. This is unusual and we don't do it anywhere else I am aware of.